src/main/java/marustuff/recipeSystem/Data/ResourceMapper.java

```diff
@@ -0,0 +1,65 @@
+package marustuff.recipeSystem.Data;
+
+import org.apache.ibatis.annotations.*;
+
+import java.util.List;
+
+@Mapper
+public interface ResourceMapper {
+
+    @Insert("INSERT INTO recipes_ingredients_amount(IngredientId,RecipeId,amount) VALUES (#{ingredientId},#{recipeId},#{amount})")
+    @Options(keyProperty = "IngredientId,RecipeId")
+    public void insertRecipeIngredientsAmount(long ingredientId, long recipeId, String amount);
+
+    @Select("SELECT Name,Id FROM recipes ORDER BY Id DESC LIMIT 15;")
+    public List<Recipe> getNameIdRecipe();
+
+    @Select("SELECT COUNT(*)/5 FROM recipes")
+    public double getNumberOfPages();
+
+    @Select("SELECT Name,Id FROM recipes ORDER BY Id LIMIT 5 OFFSET #{offset}")
+    public List<Recipe> getNameIdRecipeByPage(int offset, int currentPage);
+
+    @Select("Select Name,Id FROM recipes WHERE Name Like #{search}")
+    public List<Recipe> searchByRecipeName(String search);
+
+    @Select("Select DISTINCT recipes.Name,recipes.Id FROM  recipes LEFT JOIN (recipes_ingredients_amount  LEFT JOIN ingredients ON ingredients.id=recipes_ingredients_amount.ingredientId)ON recipes_ingredients_amount.recipeID=recipes.id WHERE ingredients.Name Like #{search};")
+    public List<Recipe> searchByIngredient(String search);
+
+    @Select("SELECT EXISTS(SELECT * FROM information_schema.tables WHERE table_schema='recipedb' AND table_name='recipes')")
+    public String isRecipesTablePresent();
+
+    @Select("SELECT EXISTS(SELECT * FROM information_schema.tables WHERE table_schema='recipedb' AND table_name='ingredients')")
+    public String isIngredientsTablePresent();
+
+    @Select("SELECT EXISTS(SELECT * FROM information_schema.tables WHERE table_schema='recipedb' AND table_name='recipes_ingredients_amount')")
+    public String isRecipesIngredientsAmountsTablePresent();
```

Comment:
powinieneś móc tutaj zwrócić wartość boolowską. String to raczej nie najlepszy pomysł